tests/acceptance/QueryBuilderDbal.feature

```diff
@@ -0,0 +1,135 @@
+Feature: QueryBuilderDbal
+  In order to use Doctrine DBAL QueryBuilder safely
+  As a Psalm user
+  I need Psalm to typecheck QueryBuilder
+
+  Background:
+    Given I have the following config
+      """
+      <?xml version="1.0"?>
+      <psalm totallyTyped="true">
+        <projectFiles>
+          <directory name="."/>
+        </projectFiles>
+        <plugins>
+          <pluginClass class="Weirdan\DoctrinePsalmPlugin\Plugin" />
+        </plugins>
+      </psalm>
+      """
+    And I have the following code preamble
+      """
+      <?php
+      use Doctrine\DBAL\Query\QueryBuilder;
+      use Doctrine\DBAL\Query\Expression\CompositeExpression;
+
+      /**
+       * @psalm-suppress InvalidReturnType
+       * @return QueryBuilder
+       */
+      function builder() {}
+      """
+
+  @QueryBuilderDbal
+  Scenario: Dbal QueryBuilder ::select accepts variadic arguments
+    Given I have the following code
+      """
+      builder()->select('field1', 'field2')->distinct();
```

Comment:
`distinct()` was [added in 2.10](https://github.com/doctrine/dbal/pull/3696), while the lowest version currently supported is 2.6